h2o-core/src/main/java/water/rapids/BinaryMerge.java

```diff
@@ -49,15 +52,19 @@
     private final long _base[]; // the col.min() of each column in the key
     private final int _fieldSizes[]; // the widths of each column in the key
     private final int _keySize; // the total width in bytes of the key, sum of field sizes
+    private final BigInteger _baseD[];  // the col.min() of each column in double
+    private final boolean _isNumeric[];
 
-    FFSB( Frame frame, int msb, int shift, int fieldSizes[], long base[] ) { 
+    FFSB( Frame frame, int msb, int shift, int fieldSizes[], long base[], BigInteger baseD[], boolean isNumeric[]) {
```

Comment:
This PR uses longs for categoricals and BigIntegers for numeric values. Can we unify this and use BigInteger for everything?

This would simplify this change a little bit (even though BigIntegers would be overkill for categoricals).